ruby/private/bundle/create_bundle_build_file.rb

```diff
@@ -252,12 +252,23 @@ def register_gem(spec, template_out, bundle_lib_paths, bundle_binaries)
     # Do not register local gems
     return if spec.source.path?
 
-    gem_path = GEM_PATH[ruby_version, spec.name, spec.version]
-    spec_path = SPEC_PATH[ruby_version, spec.name, spec.version]
-    base_dir = "lib/ruby/#{ruby_version}"
+    if spec.source.is_a?(Bundler::Source::Git)
+      stub = spec.source.specs.find { |s| s.name == spec.name }.stub
+      base_dir = "lib/ruby/#{ruby_version}"
+      gem_path = "#{base_dir}/bundler/gems/#{Pathname(stub.full_gem_path).relative_path_from(stub.base_dir)}"
+      spec_path = "#{base_dir}/bundler/gems/#{Pathname(stub.loaded_from).relative_path_from(stub.base_dir)}"
+
+      # paths to register to $LOAD_PATH
+      require_paths = stub.require_paths
```

Comment:
Move `require_paths = if ...`

Reply:
Like this?

```rb
    base_dir = "lib/ruby/#{ruby_version}"
    # paths to register to $LOAD_PATH
    require_paths = if spec.source.is_a?(Bundler::Source::Git)
                      stub = spec.source.specs.find { |s| s.name == spec.name }.stub
                      gem_path = "#{base_dir}/bundler/gems/#{Pathname(stub.full_gem_path).relative_path_from(stub.base_dir)}"
                      spec_path = "#{base_dir}/bundler/gems/#{Pathname(stub.loaded_from).relative_path_from(stub.base_dir)}"
                      stub.require_paths
                    else
                      gem_path = GEM_PATH[ruby_version, spec.name, spec.version]
                      spec_path = SPEC_PATH[ruby_version, spec.name, spec.version]
                      Gem::StubSpecification.gemspec_stub(spec_path, base_dir, "#{base_dir}/gems").require_paths
                    end
```

But, `gem_path` and `spec_path` are used after the if statement too, like `require_paths`.
Should write `xxx = if ...` style, only `require_paths` definition?

Reply:
You can return two things from this block, eg:

```ruby
require_paths, gem_paths = if

```